### ISSUE TITLE
CVE-2017-10789: mysql_ssl=1 does not enforce encryption

### ISSUE THREAD

**Issue by pali** [perl5-dbi/DBD-mysql#140]
DBD::mysql connects to MySQL server without SSL encryption even when `mysql_ssl=1` is set together with `mysql_ssl_ca_file` certificate. Test case:

```
$ perl -MData::Dumper -MDBI -e 'my $dbh = DBI->connect("dbi:mysql:mysql;host=127.0.0.1", "root", "password", { mysql_ssl => 1, mysql_ssl_ca_file => "ca.pem" }); print Dumper $dbh->selectall_arrayref("SELECT 1");'
```

Related to https://github.com/perl5-dbi/DBD-mysql/issues/110

**Comment by pali** [comment]
http://cve.mitre.org/cgi-bin/cvename.cgi?name=2017-10789

**Comment by dveeden** [comment]
This has been fixed in 4.044
